docs/spec/dataclasses.rst

```diff
@@ -526,6 +532,56 @@ This includes, but is not limited to, the following semantics:
   dataclass must be explicitly annotated as e.g. ``x: ClassVar[Final[int]] =
   3``.
 
+Converters
+^^^^^^^^^^
+
+The ``converter`` parameter can be specified in a field definition to provide a single-argument callable
+used to convert values when assigning to the associated attribute. This feature allows for automatic type
+conversion and validation during object initialization and attribute assignment.
+
+Converter behavior:
+
+* For non-frozen dataclasses, the converter is used for all attribute assignment, including assignment
+  of default values and direct attribute setting (e.g., ``obj.attr = value``).
+* For frozen dataclasses, the converter is only used inside a ``dataclass``-synthesized ``__init__``
+  when setting the attribute.
+* The converter is not used when reading attributes, as the attributes should already have been converted.
+
+Typing rules for converters:
+
+* The ``converter`` must be a callable that accepts a single positional argument.
```

Comment:
"callable that accepts a single positional argument" => "a callable with one positional-only parameter"

Reply:
Actually I'll push back on this one, since the callable can be more than a one-positional-parameter-only callable. E.g. `foo(*args)` or `foo(x, y=None)` should be compatible.

Reply:
(although I think this is more of a problem _most_ places we say single-parameter. Maybe it's worth clarifying once at the top, that by "single-parameter" we mean "single-parameter-compatible"? (or just use that slightly verbose wording throughout?)

Reply:
Let's stay away from "compatible" -- that word isn't defined to have any particular meaning in the spec.

I think the current wording here ("callable that accepts a single positional argument") is OK, and accurately describes the requirements, without placing additional unnecessary restrictions on the callable.

Unless @erictraut you were intending to suggest a more substantive change, and you believe the callable really should be restricted to only accepting one parameter, and that parameter must be positional-only? I'm not sure I see the rationale for those limitations.

Reply:
Above, the spec talks about a "single-parameter callable", so I was trying to make the language consistent here and use "parameter" rather than "argument", but in the process I unintentionally added a new limitation. I agree that limiting it to a positional-only parameter is unnecessary. The current wording is fine.

Reply:
Yeah this is exactly the kind of ambiguity I'm hoping to avoid by being explicit 😂

I'm happy to tweak the wording though. Would "assignable" be a correct thing to say?

Alternatively something akin to "must accept a single positional argument" (which allows any parameter spec that matches the criteria)